accounts/abi/abigen/source2.go.tpl

```diff
@@ -93,12 +93,8 @@ var (
 		// the contract method with ID 0x{{printf "%x" .Original.ID}}.
 		//
 		// Solidity: {{.Original.String}}
-		func ({{ decapitalise $contract.Type}} *{{$contract.Type}}) Pack{{.Normalized.Name}}({{range .Normalized.Inputs}} {{.Name}} {{bindtype .Type $structs}}, {{end}}) []byte {
-			enc, err := {{ decapitalise $contract.Type}}.abi.Pack("{{.Original.Name}}" {{range .Normalized.Inputs}}, {{.Name}}{{end}})
```

Comment:
I think that the reason we chose to panic here is that any error arising from the pack call here indicates programmer error:  e.g. some parameter was nil when the calling code should have validated it before trying to pack.

My feeling is that if we adopt the changes in this PR, we should leave the semantics of this method unchanged and add a `TryPack` method.

Reply:
Great idea to add the TryPack method, in my opinion, as the basic tool library, it directly determines whether panic is not appropriate.

Reply:
The `TryPack` method is added in this [commit](https://github.com/ethereum/go-ethereum/pull/31692/commits/4c524e6b1db9bca7efaf3a7612f24ed0141d6670).